_blog-src/_posts/2015-06-15-apiman-msa-day.adoc

```diff
@@ -0,0 +1,60 @@
+
+////
+Use this pattern to avoid asciidoctor rendering the HTML comment.
+<!--more-->
+////
+
+== Services, services, everywhere!
+
+A microservices architecture encourages the decomposition of problems into small, isolated and reusable chunks; much in the same spirit as UNIX's _do one job and do it well_ philosophy. This brings with it a whole raft of benefits, for instance: we can scale each component independently, avoiding unnecessarily dragging along other components which just happen to reside in the same monolith; we can easily have a single team wholly responsible for a single microservice's lifecycle, simplifying debugging and allowing them to use their skills and technologies of choice; and, we can easily create new functionality via composition. However, this proliferation of heterogeneous services poses some challenges, a significant subset of which API Management is a key solution to.
+
+== Avoiding reimplementation of cross-cutting functionality
+
+A problem that can quickly become apparent is that there are often key pieces of important functionality that is difficult or undesirable to reimplement in every API. Good examples of this would be security concerns like authentication and authorization, or business concerns like rate-limiting, billing and auditing.
+
+It's quite clear why we would want to avoid this situation. Imagine we have hundreds of microservices we want to expose to the outside world; the nightmare scenario is that we have to implement complex and prone-to-change authentication and authorization schemes on _each_ of them. Quite aside from bloating and complicating our microservices, it is going to be extremely difficult to guarantee that each service has implemented all of this common functionality consistently and correctly - we could quite plausibly be leaving holes in our system simply by virtue of surface area.
+
+Instead, we can drastically improve the situation by using apiman to relieve individual services of that burden; simply configure and manage those tasks in a single, well-known location. We register our APIs with apiman, apply appropriate policies, and route public traffic through the gateway. We leave apiman to deal with these common problems, and let the microservice's developers focus on their core task, rather than burning time on ancillary concerns.
+
+== Fickle business requirements
+
+Access to APIs often has business requirements which are independent of the core functionality service itself offers. For instance, whilst a mapping application might only be interested in coordinates, its business requirements could dictate that we need to capture metrics; initiate contracts with, and issue keys to, API consumers; and, audit their interactions with the system. Aside the microservices _faux pas_ of implementing external business logic, we would want to avoid this because business requirements tend to change rapidly in response to events which are unrelated to the service itself.
```

Comment:
Changed this to

```asciidoc
Access to APIs often has business requirements which are independent of the core functionality the service itself offers. For instance, whilst a mapping application might only be interested in coordinates, its business requirements could dictate that we need to capture metrics; initiate contracts with, and issue keys to, API consumers; and, audit their interactions with the system. Quite aside from the microservices _faux pas_ of contaminating a service with external business logic, we generally want to avoid this situation, as business requirements tend to change rapidly in response to events which are unrelated to the service's purpose.
```